module-system/module-implementations/examples/sov-election/src/call.rs

```diff
@@ -7,12 +7,16 @@ use anyhow::{anyhow, bail, ensure, Result};
 use sov_modules_api::{CallResponse, Context};
 use sov_state::WorkingSet;
 
+#[cfg(feature = "native")]
+use sov_modules_macros::CustomParser;
+
 /// Call actions supported byte the module.
 #[cfg_attr(
     feature = "native",
     derive(serde::Serialize),
     derive(serde::Deserialize)
 )]
+#[cfg_attr(feature = "native", derive(CustomParser), module_name = "Election")]
```

Comment:
Would it be possible to add `fn module_name()` to the `ModuleInfo` trait and have the name generated automatically? We could use the same approach for `RPC`

Reply:
hmm yeah, that's a good idea. let me look into that. the main reason i passed the module_name is because we didnt have access to it, but we can build that into the trait

Reply:
@bkolad If we add a `fn module_name()` to the `ModuleInfo`, we still won't have access to it in this context right? Or am I missing something?